tests/param/cast_request_param_test.py

```
@@ -11,9 +17,29 @@ def test_empty_string_becomes_none_for_type_integer():
     assert cast_request_param('integer', 'biz_id', '') is None
 
 
-def test_boolean():
-    assert cast_request_param('boolean', 'is_open', 1)
-    assert not cast_request_param('boolean', 'is_open', 0)
+def test_boolean_true_is_true_or_1():
+    assert cast_request_param('boolean', 'is_open', 'true')
+    assert cast_request_param('boolean', 'is_open', 'tRUe')
+    assert cast_request_param('boolean', 'is_open', '1')
+
+
+def test_boolean_false_is_false_or_0():
+    assert not cast_request_param('boolean', 'is_open', 'false')
+    assert not cast_request_param('boolean', 'is_open', 'faLSe')
+    assert not cast_request_param('boolean', 'is_open', '0')
+
+
+@patch('bravado_core.param.log')
+def test_boolean_returns_unchanged_string_for_non_bool_strings(mock_logger):
+    assert cast_request_param('boolean', 'is_open', 'PIZZA') == 'PIZZA'
+    assert not cast_request_param('boolean', 'is_open', 'PIZZA') is True
+    assert not cast_request_param('boolean', 'is_open', 'PIZZA') is False
+
+
+@patch('bravado_core.param.log')
+def test_boolean_cast_failure_is_logged(mock_logger):
```

I'll have this test call the function directly and check that a `ValueError` is raised rather than checking by proxy via the logging behavior.